tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -247,44 +254,53 @@ internal static bool TryParseTraceParent(string header, out W3CTraceParent trace
 #if NETCOREAPP
             var w3cTraceId = header.AsSpan(start: 3, length: 32);
             var w3cSpanId = header.AsSpan(start: 36, length: 16);
-            traceId = ParseUtility.ParseFromHexOrDefault(w3cTraceId[16..]);
 
-            if (traceId == 0)
+            if (!HexString.TryParseUInt64(w3cTraceId[16..], out traceId) || traceId == 0)
             {
                 return false;
             }
 
-            parentId = ParseUtility.ParseFromHexOrDefault(w3cSpanId);
-
-            if (parentId == 0)
+            if (!HexString.TryParseUInt64(w3cSpanId, out parentId) || parentId == 0)
             {
                 return false;
             }
 
             rawTraceId = w3cTraceId.ToString();
             rawSpanId = w3cSpanId.ToString();
+            bool sampled;
 
-            var traceFlags = header.AsSpan(start: 53, length: 2);
-            var sampled = (ParseUtility.ParseFromHexOrDefault(traceFlags) & 1) == 1;
+            if (HexString.TryParseByte(header.AsSpan(53, 2), out var traceFlags))
+            {
+                sampled = ((TraceFlags)traceFlags & TraceFlags.Sampled) == TraceFlags.Sampled;
```

Comment:
Just noting that if you Added `[EnumExtensions]` to the `[TraceFlags]` enum, you could rewrite this as:

```suggestion
                sampled = traceFlags.HasFlag(TraceFlags.Sampled);
```

up to you whether you want to or not - it's easier to read IMO, but prob not worth it while we only have a single value

Reply:
Thanks. Done, although I had to make `TraceFlags` internal (a nested private enum causes compiler errors in the generated code), and still had to cast the `byte`:
```csharp
sampled = ((TraceFlags)traceFlags).HasFlagFast(TraceFlags.Sampled);
```

Reply:
> Done, although I had to make TraceFlags internal

Ah, good point 🤔 I'm not sure there's any way round that. Like I say, maybe not worth it (every time we add `[EnumExtensions]` to an enum, the IDE has to do more work 😅), was just for information really